Repository/CmfEditableRepository.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2015 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Component\Resource\Repository;
+
+use InvalidArgumentException;
+use Puli\Repository\Api\EditableRepository;
+use Puli\Repository\Api\UnsupportedLanguageException;
+
+/**
+ * CMF own interface to add the move() method.
+ *
+ * @author Maximilian Berghoff <Maximilian.Berghoff@mayflower.de>
+ */
+interface CmfEditableRepository extends EditableRepository
+{
+    /**
+     * Moves a resource inside the repository.
+     *
+     * @param string $sourceQuery The Path of the current document.
+     * @param string $targetPath  The parent path of the destination.
+     * @param string $language
+     *
+     * @return int
+     *
+     * @throws InvalidArgumentException     If the sourceQuery is invalid.
+     * @throws InvalidArgumentException     If the resource can not be moved to the targetPath.
+     * @throws UnsupportedLanguageException If the language is not supported.
+     */
+    public function move($sourceQuery, $targetPath, $language = 'glob');
```

Comment:
source path would make more sense maybe. what with the language =
'glob'? is the idea that we allow to split languages? i would not allow
that, its rather specific. maybe instead have a method to delete one
language? then the flow would be copy and delete some languages from old
and new document.